platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/JettyAuthenticator.java

```diff
@@ -110,17 +107,50 @@ public void doFilter(
             bundleContext.getService(securityFilterServiceReference);
 
         if (!hasBeenInitialized(securityFilterServiceReference, bundleContext)) {
-          initializeSecurityFilter(bundleContext, securityFilterServiceReference, securityFilter);
+          try {
+            initializeSecurityFilter(
+                bundleContext, securityFilterServiceReference, securityFilter, resolveRequest(req));
+          } catch (ServletException e) {
+            throw new ServerAuthException("Unable to initialize security filter");
+          }
         }
-
         chain.addSecurityFilter(securityFilter);
       }
 
-      chain.doFilter(servletRequest, servletResponse);
+      try {
+        chain.doFilter(req, res);
+      } catch (IOException | ServletException e) {
+        throw new ServerAuthException(
+            "Unable to process security filter. Blocking the request processing.");
+      }
     } else {
       LOGGER.debug("Did not find any SecurityFilters. Acting as a pass-through filter...");
-      filterChain.doFilter(servletRequest, servletResponse);
     }
+
+    Subject subject = (Subject) req.getAttribute(SecurityConstants.SECURITY_SUBJECT);
+    UserIdentity userIdentity = new JettyUserIdentity(getSecuritySubject(subject));
+    if (userIdentity == null) return null;
+
+    return new JettyAuthenticatedUser(userIdentity);
+  }
+
+  BundleContext getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(JettyAuthenticator.class);
+    if (cxfBundle != null) {
+      return cxfBundle.getBundleContext();
+    }
+    return null;
+  }
+
+  protected Request resolveRequest(ServletRequest req) {
+    return (req instanceof Request) ? (Request) req : null;
+  }
+
+  private javax.security.auth.Subject getSecuritySubject(Subject subject) {
+    HashSet emptySet = new HashSet();
+    HashSet subjectPrincipal = new HashSet();
+    subjectPrincipal.add(subject);
+    return new javax.security.auth.Subject(true, subjectPrincipal, emptySet, emptySet);
```

Comment:
Unless I'm mistaking what is going on here, we need to do a little more when building this Java Subject object. If you take a look at the LoginFilter `doFilter` method, you'll see that we actually build one of these types of Subjects at that point and attach it to the thread. That subject contains the name of the Subject and all of the Roles as principals. That code on the LoginFilter shouldn't actually be doing anything anymore, since we're now using this Subject instead. So what I think we need to do, is remove the code from the LoginFilter and build all of the principals here. So we'd need to do ```SecurityAssertion securityAssertion =
                  subject.getPrincipals().oneByType(SecurityAssertion.class);
subjectPrincipal.addAll(securityAssertion.getPrincipals())``` after adding the Subject itself, like you're already doing. Without those extra role principals, some things can fail to work correctly down in the Karaf code when doing admin operations.